examples/src/js/PropsDemoExample.js

```diff
@@ -0,0 +1,417 @@
+        default: true,
+    },
+    {
+        label: 'showExpandAll',
+        value: 'showExpandAll',
+        checked: false,
+        default: false,
+    },
+    {
+        label: 'showNodeIcon',
+        value: 'showNodeIcon',
+        checked: true,
+        default: true,
+    },
+    {
+        label: 'shodeNodeTitle',
+        value: 'shodeNodeTitle',
+        checked: false,
+        default: false,
+    },
+];
+
+class PropsDemoExample extends React.Component {
+    state = {
+        clicked: {
+            value: 'nothing yet',
+        },
+        nodes: initialNodes,
+        checkboxParams: initialParams,
+    };
+
+    onCheck = (node, nodes) => {
+        this.setState({ nodes });
+    }
+
+    onClick = (clicked) => {
+        // console.log(`clicked = ${clicked.value}`);
+        this.setState({ clicked });
+    }
+
+    onExpand = (node, nodes) => {
+        this.setState({ nodes });
+    }
+
+    onParameterChange = (param, params) => {
+        this.setState({ checkboxParams: params });
+    }
+
+    getParams = () => {
+        const { checkboxParams } = this.state;
+        const params = {};
+        checkboxParams.forEach((param) => {
+            if (!param.radioGroup) {
+                params[param.value] = param.checked || false;
+            } else if (param.checked) {
+                param.children.forEach((child) => {
+                    if (child.checked) {
+                        params[param.value] = child.value;
+                    }
+                });
+            }
+        });
+        return params;
+    }
+
+    restoreDefaultParams = () => {
+        const { checkboxParams } = this.state;
+        const newParams = [];
+        checkboxParams.forEach((param) => {
+            const newParam = { ...param };
+            newParam.checked = param.default || false;
+            if (param.radioGroup) {
+                const newChildren = param.children.map((child) => {
+                    const newChild = { ...child };
+                    newChild.checked = newChild.default || false;
+                    return newChild;
+                });
+                newParam.children = newChildren;
+            }
+            newParams.push(newParam);
+        });
+        this.setState({ checkboxParams: newParams });
+    }
+
+    render() {
+        const {
+            checkboxParams,
+            clicked,
+            nodes,
+        } = this.state;
+
+        const style3 = {
+            width: '30%',
+            margin: '5px',
+            border: '1px solid green',
+            padding: '5px 0px 5px 5px',
+        };
+
+        const params = this.getParams();
+
+        // to test "expandOnClick"
+        let clickHandler;
+        if (params.expandOnClick) {
+            clickHandler = this.onClick;
+        }
+
+        // console.log(params);
+        // console.log('------------------------------------------------');
+
+        return (
+            <div style={{ display: 'flex' }}>
```

Comment:
I can follow your lead on inline styles. I mostly use them for testing as it is much quicker for me.

This comprehensive example has been very important to me for testing.  I placed it at the top of the examples because it was the one I was using the most to test changes. I find it helpful in finding out the effect of a prop change and what is the effect of the various different combinations of props.

By changing the props interactively, the effects of the change are quickly shown as opposed to creating a new simple example with a particular combination of props. I did have to make several changes due to problems found when mixing props in different combinations.

You may want to put this example last behind the simpler examples to be the last one a user may visit.